### PR TITLE
Add package licensing guide

### PR DESCRIPTION
**Pull request by KeyWeeUsr**
I found out there's a damn lot of things you can miss or misread while licensing a binary package. I made this guide as a request for other info mainly related to binaries created on/for Mac/iOS (don't have an access to those machines), so that this guide could be completed. As I mentioned in the warning, the guide has only an informative character, kind of check-list of what to do before releasing into public.

Licensing may not seem important to users who make small apps or basically only learn how to code with Kivy, however if used in a serious production _and_ if licensed badly or with some stuff missing, it may shine a bad light at Kivy as a whole.

Basically I'd need some diff of source <-> binary i.e. what does a packager include to a user's source for each missing platform, so that I could find a license and paste it there. If you find the files and licenses, make a commit (devs) or a comment (who can't / don't want to).

### CODE REVIEW

File: doc/sources/guide/licensing.rst [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+Your code alone may not require including licensing info and copyright notices
+of other used software, but binaries are something else. When a binary (.exe,
+.app, .apk, ...) is created it includes Kivy, its dependencies and other
+packages that your application uses. Some of them are licensed in a way they
+require including a copyright notice somewhere in your app (or more). Before
+distributing any of the binaries, please **check all the created files** that
+don't belong to your source (.dll, .pyd, .so, ...) and include approperiate
+copyright notices if required by the license the files belong to. This way you
+should satisfy licensing requirements of the Kivy deps.
```
Comment (matham) [inline doc/sources/guide/licensing.rst]: Instead of should, `may` is probably better.

File: doc/sources/guide/licensing.rst [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+* `docutils <https://sourceforge.net/p/docutils/code/HEAD/tree/trunk/docutils/COPYING.txt>`_
+* `pygments <https://bitbucket.org/birkenfeld/pygments-main/src/a042025b350cd9c9461f7385d9ba0f13cdb01bb9/LICENSE>`_
+* `sdl2 <https://www.libsdl.org/license.php>`_
+* `glew <http://glew.sourceforge.net/glew.txt>`_
+* `gstreamer <https://github.com/GStreamer/gstreamer/blob/master/COPYING>`_
+  (if used)
+* image & audio libraries(e.g. |mixer|_)
+
+You'll probably need to check image and audio libraries manually (mostly
+begin with ``lib``). The ``LICENSE*`` files that belong to them should be
```
Comment (matham) [inline doc/sources/guide/licensing.rst]: Does pyinstaller look for license files?

File: doc/sources/guide/licensing.rst [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+included by PyInstaller, but are not by python-for-android and you need
+to find them.
+
+Windows (PyInstaller)
+---------------------
+
+.. |win32| replace:: pypiwin32
+.. _win32: https://pypi.python.org/pypi/pypiwin32
+
+To access Windows API, Kivy uses |win32|_ package. This package is released
```
Comment (matham) [inline doc/sources/guide/licensing.rst]: To access [some] Windows API [features].

File: doc/sources/guide/licensing.rst [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+.. |py2crt| replace:: Py2 CRT license
+.. _py2crt: https://hg.python.org/sandbox/2.7/file/tip/Tools/msi/crtlicense.txt
+.. |py3crt| replace:: Py3 CRT license
+.. _py3crt: https://hg.python.org/cpython/file/tip/Tools/msi/exe/crtlicense.txt
+.. |redist| replace:: List of redistributables
+.. _redist: https://msdn.microsoft.com/en-us/library/8kche8ah(v=vs.90).aspx
+
+Python compiled with Visual Studio (official) has some files from Microsoft
+and you are required to protect them if you distribute them. You can do that
```
Comment (matham) [inline doc/sources/guide/licensing.rst]: Protect seems a bit out of place.

Comment (matham) [inline doc/sources/guide/licensing.rst]: You can do that -> Including names...may satisfy that requirement.

File: doc/sources/guide/licensing.rst [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+Missing.
+
+Avoiding binaries
+-----------------
+
+.. |cons| replace:: consequences
+.. _cons: http://programmers.stackexchange.com/a/234295
+
+There is a way how to avoid this licensing process with avoiding creating
```
Comment (matham) [inline doc/sources/guide/licensing.rst]: This should also be stated, rather than authoritative, it should be more suggestive.

File: doc/sources/guide/licensing.rst (hunk 2 above)
Comment (KeyWeeUsr) [inline doc/sources/guide/licensing.rst]: nope, those are copied only because of the kivy.deps hook, because SDL2 (or Kivy in kivy.deps.sdl2) has the LICENSE* files distributed together with binaries. If you have an access to Win, check wheel installation and `<python>/share/sdl2/bin` folder

Comment (matham) [inline doc/sources/guide/licensing.rst]: Oh, yeah, I included them there :D

**Comment by matham** [comment]
Feel free to merge if it's ready.

**Comment by KeyWeeUsr** [comment]
There's still a hole in the mac and ios part, but I don't have an access to those, so I guess it can be merged in this state rather than just sit here incomplete. Hopefully someone will add those missing parts. I have a feeling it'll be similar to linux part.
